### PR TITLE
rewriting, refactoring and implementation of latest ut port version

### PR DESCRIPTION
**Pull request by Mitko-G**

### CODE REVIEW

**Comment by kalinkrustev** [comment]
Please check the jenkins errors

**Comment by Mitko-G** [comment]
Errors are fixed

File: index.js [hunk 1]
```diff
@@ -1,267 +1,260 @@
...
-    this.latency = this.counter && this.counter('average', 'lt', 'Latency');
-
-    if (this.config.protocol === 'sftp') {
-        FtpClient = require('scp2/lib/client').Client;
-    } else {
-        FtpClient = require('ftp');
+module.exports = function({parent}) {
+    function FtpPort({config}) {
+        parent && parent.apply(this, arguments);
+        this.config = Object.assign({
```
Comment (kalinkrustev) [inline index.js]: We have to use lodash.merge when the object we merge into is not flat. 
For example doing
```js
Object.assign({host:'localhost',port:22}, {port: 1000})
```
will overwrite the default host with undefined host.

**Review by kalinkrustev — CHANGES_REQUESTED** [review]

**Comment by Mitko-G** [comment]
done

**Pull request merged**